File: 3 КУРС/2 семестр/Java/JavaProects2/app/src/main/java/com/example/voiceassistent/MessageListAdapter.java
package com.example.voiceassistent;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.RecyclerView.Adapter;

import java.util.ArrayList;
import java.util.List;

public class MessageListAdapter extends Adapter {
    public List<Message> messageList = new ArrayList<>();
    private static final int ASSISTANT_TYPE=0;
    private static final int USER_TYPE=1;

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = null;
        if (viewType == USER_TYPE) {
            view = LayoutInflater.from(parent.getContext())
                    .inflate(R.layout.user_message,parent,false);

            //создание сообщения от пользователя
        }
        else {
            view = LayoutInflater.from(parent.getContext())
                    .inflate(R.layout.assistant_message,parent,false);
            //создание сообщения от ассистента
        }

        assert view != null;
        return new MessageViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        new MessageViewHolder(holder.itemView).bind(messageList.get(position));

    }

    public int getItemViewType(int index){
        Message message = messageList.get(index);
        if (message.isSend){
            return USER_TYPE;
        }
        else return ASSISTANT_TYPE;
    }


    @Override
    public int getItemCount() {
        return messageList.size();

    }
}

File: 3 КУРС/2 семестр/Учебная практика/JavaProects/app/src/main/java/com/example/voiceassistent/MessageListAdapter.kt
package com.example.voiceassistent

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import java.util.*
import com.example.voiceassistent.R


class MessageListAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    var messageList: ArrayList<Message> = ArrayList()


    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): RecyclerView.ViewHolder {
        val view: View
        return if (viewType == USER_TYPE) {
            view = LayoutInflater.from(parent.context)
                .inflate(R.layout.user_message, parent, false)
            MessageViewHolder(view)
        } else {
            view = LayoutInflater.from(parent.context)
                .inflate(R.layout.assistant_message, parent, false)
            MessageViewHolder(view)
        }
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        MessageViewHolder(holder.itemView).bind(messageList[position])
    }
    override fun getItemCount(): Int {
        return messageList.size

    }

    override fun getItemViewType(index: Int): Int {
        var message = messageList[index]
        return if (message.isSend) {
            USER_TYPE
        } else ASSISTANT_TYPE
    }

    companion object {
        private const val ASSISTANT_TYPE = 0
        private const val USER_TYPE = 1
    }

}